src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Login/LoginHandler.cs

```diff
@@ -82,10 +82,19 @@ public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRe
                 ConnectPermission = prototype.WindowsGrantAccess,
                 IsEnabled = !prototype.IsDisabled,
                 IsLockedOut = prototype.IsLockedOut,
-                UserMapping = new ServerLoginDatabaseUserMapping[0],
-                SecurablePermissions = prototype.SecurablePermissions
+                UserMapping = new ServerLoginDatabaseUserMapping[0]
             };
 
+            // show permissions for new logins and existing non-system logins
+            if (!prototype.HidePermissions && (dataContainer.IsNewObject || !((dataContainer.Server.GetSmoObject(parameters.ObjectUrn) as Login).IsSystemObject)))
```

Comment:
it looks like a Login is being created in a similar condition on line 59 above, and cached in the state object in the prototype.  It may be cheaper to either cache the instance created above in this method, or to add an accessor to the one stored in the state in the prototype.  I'm assuming that GetSmoObject is running SQL queries in order to create the Login, if that's not the case than I don't think it matters as much.

Reply:
Re-used the login object